File: src/main/java/com/github/trcdeveloppers/clayium/Clayium.java
package com.github.trcdeveloppers.clayium;

import com.github.trcdeveloppers.clayium.blocks.machines.clayworktable.ClayWorkTableRecipes;
import com.github.trcdeveloppers.clayium.blocks.machines.clayworktable.TileClayWorkTable;
import com.github.trcdeveloppers.clayium.gui.GuiHandler;
import com.github.trcdeveloppers.clayium.proxy.ClayiumCommonProxy;
import com.github.trcdeveloppers.clayium.worldgen.ClayOreGenerator;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.SidedProxy;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.fml.common.network.NetworkRegistry;
import net.minecraftforge.fml.common.registry.GameRegistry;

@Mod(
    modid = Clayium.MOD_ID,
    name = Clayium.MOD_NAME,
    version = Clayium.VERSION
)
public class Clayium {

    public static final String MOD_ID = "clayium";
    public static final String MOD_NAME = "Clayium";
    public static final String VERSION = "1.0-SNAPSHOT";

    /**
     * This is the instance of your mod as created by Forge. It will never be null.
     */
    @Mod.Instance(MOD_ID)
    public static Clayium INSTANCE;

    @SidedProxy(clientSide = "com.github.trcdeveloppers.clayium.proxy.ClayiumClientProxy",
                serverSide = "com.github.trcdeveloppers.clayium.proxy.ClayiumCommonProxy")
    public static ClayiumCommonProxy proxy;

    /**
     * This is the first initialization event. Register tile entities here.
     * The registry events below will have fired prior to entry to this method.
     */
    @Mod.EventHandler
    public void preinit(FMLPreInitializationEvent event) {
        MinecraftForge.EVENT_BUS.register(proxy);
        GameRegistry.registerWorldGenerator(new ClayOreGenerator(), 0);

        GameRegistry.registerTileEntity(TileClayWorkTable.class, new ResourceLocation(MOD_ID, "TileClayWorkTable"));

        NetworkRegistry.INSTANCE.registerGuiHandler(this, new GuiHandler());
    }

    /**
     * This is the second initialization event. Register custom recipes
     */
    @Mod.EventHandler
    public void init(FMLInitializationEvent event) {
        proxy.registerItemColors();
        ClayWorkTableRecipes.init();
    }

    /**
     * This is the final initialization event. Register actions from other mods here
     */
    @Mod.EventHandler
    public void postinit(FMLPostInitializationEvent event) {

    }
}

File: src/main/kotlin/com/github/trcdeveloppers/clayium/Clayium.kt
package com.github.trcdeveloppers.clayium

import com.github.trcdeveloppers.clayium.common.ClayiumCommonProxy
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.SidedProxy
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

@Mod(modid = Clayium.MOD_ID, name = Clayium.MOD_NAME, version = Clayium.VERSION)
class Clayium {
    @Mod.EventHandler
    fun preInit(event: FMLPreInitializationEvent) {
        // If proxy is null, something is wrong.
        proxy!!.preInit(event)
    }

    @Mod.EventHandler
    fun init(event: FMLInitializationEvent) {
        proxy!!.init(event)
    }

    @Mod.EventHandler
    fun postInit(event: FMLPostInitializationEvent) {
        proxy!!.postInit(event)
    }

    companion object {
        const val MOD_ID = "clayium"
        const val MOD_NAME = "Clayium"
        const val VERSION = "1.0-SNAPSHOT"

        @JvmField
        val LOGGER: Logger = LogManager.getLogger(MOD_ID)
        /**
         * This is the instance of your mod as created by Forge. It will never be null.
         */
        @JvmField
        @Mod.Instance(MOD_ID)
        var INSTANCE: Clayium? = null

        @SidedProxy(clientSide = "com.github.trcdeveloppers.clayium.client.ClayiumClientProxy", serverSide = "com.github.trcdeveloppers.clayium.common.ClayiumCommonProxy")
        var proxy: ClayiumCommonProxy? = null
    }
}
